packages/nextjs/src/config/loaders/rollup.ts

```diff
@@ -0,0 +1,110 @@
+  //     bundle as the basis for the relative path calculation, but both our temporary file and the page being wrapped
+  //     live in `pages/xyz/`, and they're the only two files in the bundle, so `pages/xyz/`` should be used as the
+  //     root. Unclear why it's not.)
+  //   - As a result of the miscalculation, our proxy module will include `export { helperFunc } from '../utils/helper'`
+  //     rather than the expected `export { helperFunc } from '../../utils/helper'`, thereby causing a build error in
+  //     nextjs..
+  //
+  // It's not 100% clear why, but telling it not to do the conversion back from absolute to relative (by setting
+  // `makeAbsoluteExternalsRelative` to `false`) seems to also prevent it from going from relative to absolute in the
+  // first place, with the result that the path remains untouched (which is what we want.)
+  makeAbsoluteExternalsRelative: false,
+});
+
+const rollupOutputOptions: RollupOutputOptions = {
+  format: 'esm' as const,
+
+  // Don't create a bundle - we just want the transformed entrypoint file
+  preserveModules: true,
+};
+
+/**
+ * Use Rollup to process the proxy module file (located at `tempProxyFilePath`) in order to split its `export * from
+ * '<wrapped file>'` call into individual exports (which nextjs seems to need).
+ *
+ * @param tempProxyFilePath The path to the temporary file containing the proxy module code
+ * @param resourcePath The path to the file being wrapped
+ * @returns The processed proxy module code, or undefined if an error occurs
+ */
+export async function rollupize(tempProxyFilePath: string, resourcePath: string): Promise<string | undefined> {
+  let finalBundle;
+
+  try {
+    const intermediateBundle = await rollup(getRollupInputOptions(tempProxyFilePath, resourcePath));
+    finalBundle = await intermediateBundle.generate(rollupOutputOptions);
+  } catch (err) {
+    __DEBUG_BUILD__ &&
+      logger.warn(
+        `Could not wrap ${resourcePath}. An error occurred while processing the proxy module template:\n${err}`,
+      );
+    return undefined;
+  }
+
+  // The module at index 0 is always the entrypoint, which in this case is the proxy module.
+  let { code } = finalBundle.output[0];
+
+  // Rollup does a few things to the code we *don't* want. Undo those changes before returning the code.
+
+  // If the wrapped file only has a default export, our `export * from '...'` won't have anything in it (`export *`
+  // never includes the default export, which must be handled separately). In order to preserve whatever side effects
+  // running the file which is the source of the exports might have, though, Rollup will still emit an `import '...'
+  // statement. We're importing the same file later on in the proxy template, so we don't need this.
+  const sideEffectsImportRegex = new RegExp("import '.*';", 'g');
+  code = code.replace(sideEffectsImportRegex, '');
```

Comment:
I'm curious, does it actually hurt to have the `import 'blah';` in the output?

- If no, wdyt about removing this code section just to reduce code complexity on our end?
- If yes, should we add to the comment what goes wrong if we have it.

Reply:
I did some testing, and it turns out it's a moot point (so I'll take it out). Originally I was requiring (rather than importing) the wrapped module, because in some previous loader that had been helpful (I forget why). In any case, it turns out that these rogue `import 'wrappedModule'` statements only show up when using `require`. In that case, Rollup can't combine the `export * from 'wrappedModule'` and the `require('wrappedModule')`, and if the `export *` ends up empty, it turns into `import 'wrappedModule'`. When the wrapped module is instead imported in the template, it _can_ combine them, so you don't get the lingering `import`. Here's the difference in the outputs (the `require` case on the left, and the `import` case on the right):

![image](https://user-images.githubusercontent.com/14812505/185525857-2d3d47bb-ed79-487c-bfa3-6533e5514831.png)